File: lib/constraints.py
from .bases import Base


class Constraint(Base):
    def __init__(self, value) -> None:
        self.value = value

    @property
    def name(self):
        return super().name.replace("_", " ")

    def __str__(self) -> str:
        return f"{self.value} {self.name}"


class NOT_NULL(Constraint):
    ...


class PRIMARY_KEY(Constraint):
    ...


class UNIQUE(Constraint):
    ...


class FOREIGN_KEY(Constraint):
    ...


class CHECK(Constraint):
    ...


class DEFERRABLE(Constraint):
    ...


class NOT_DEFERRABLE(Constraint):
    ...


class INITIALLY_IMMEDIATE(Constraint):
    ...


class INITIALLY_DEFERRED(Constraint):
    ...

File: prmp_sql_dart/constraints.dart
import 'bases.dart';

class Constraint extends Base {
  dynamic value;
  Constraint(this.value);

  @override
  String get name => super.name.replaceAll("_", " ");

  @override
  String toString() => "$value $name";
}

class NOT_NULL extends Constraint {
  NOT_NULL(value) : super(value);
}

class PRIMARY_KEY extends Constraint {
  PRIMARY_KEY(value) : super(value);
}

class UNIQUE extends Constraint {
  UNIQUE(value) : super(value);
}

class FOREIGN_KEY extends Constraint {
  FOREIGN_KEY(value) : super(value);
}

class CHECK extends Constraint {
  CHECK(value) : super(value);
}

class DEFERRABLE extends Constraint {
  DEFERRABLE(value) : super(value);
}

class NOT_DEFERRABLE extends Constraint {
  NOT_DEFERRABLE(value) : super(value);
}

class INITIALLY_IMMEDIATE extends Constraint {
  INITIALLY_IMMEDIATE(value) : super(value);
}

class INITIALLY_DEFERRED extends Constraint {
  INITIALLY_DEFERRED(value) : super(value);
}
